### PR TITLE
add mish op.

### PR DESCRIPTION
**Pull request by heavengate**
### PR types
Others

### PR changes
APIs

### Describe

**add mish op**
- add `fluid.layers.mish`
mish1
mish2

CN doc: https://github.com/PaddlePaddle/FluidDoc/pull/2152

### CODE REVIEW

File: paddle/fluid/operators/mish_op.cu [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+template <typename T>
+__global__ void KeMishBw(const T* in, const T* dout, T* din, const int numel,
+                         const float threshold) {
+  int tid = blockIdx.x * blockDim.x + threadIdx.x;
+  int stride = blockDim.x * gridDim.x;
+  for (; tid < numel; tid += stride) {
+    T x = in[tid];
+    T sp = CalcSoftplus<T>(x, threshold);
+    T tsp = tanh(sp);
+    T grad_sp = -expm1f(-sp);
```
Comment (qingqing01) [inline paddle/fluid/operators/mish_op.cu]: `expm1f`这个函数可以用于double吗？

File: paddle/fluid/operators/mish_op.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+namespace operators {
+
+using Tensor = framework::Tensor;
+
+template <typename T>
+HOSTDEVICE T CalcSoftplus(T x, float threshold) {
+  if (threshold > 0 && x > threshold) {
+    return x;
+  } else if (threshold > 0 && x < -threshold) {
+    return expf(x);
```
Comment (qingqing01) [inline paddle/fluid/operators/mish_op.h:36]: expf是用于float的吧？ 如果T是double，这里不会错？

File: paddle/fluid/operators/mish_op.cu (hunk 1 above)
Comment (heavengate) [reply paddle/fluid/operators/mish_op.cu]: Done, thanks!

File: paddle/fluid/operators/mish_op.h (hunk 2 above)
Comment (heavengate) [reply paddle/fluid/operators/mish_op.h:36]: Done, thanks!

**Review by qingqing01 — DISMISSED** [review]
LGTM.

**Review by qingqing01 — DISMISSED** [review]

**Review by qingqing01 — DISMISSED** [review]

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -14717,6 +14718,81 @@ def hard_swish(x, threshold=6.0, scale=6.0, offset=3.0, name=None):
...
+	out = \\begin{cases}
+		x \\ast \\tanh(x), \\text{if } x > \\text{threshold} \\\\
+		x \\ast \\tanh(e^{x}), \\text{if } x < -\\text{threshold} \\\\
+		x \\ast \\tanh(\\ln(1 + e^{x})),  \\text{otherwise}
+	      \\end{cases}
+
+    Args:
+        x (Variable): Input feature, multi-dimensional Tensor. The data type
+                      should be float16, float32 or float64.
+        threshold (float|None): threshold for softplus in Mish operator.
```
Comment (Heeenrrry) [inline python/paddle/fluid/layers/nn.py:14815]: 这句有些都不懂，这样改是不是符合你的原意？Approximate value of softplus will be used if absolute value
                of input is greater than :attr:`threshold` and :attr:`threshold`
                is set as positive value. For none or negative threshold,
                approximate value is not used. Default 20.

Comment (heavengate) [reply python/paddle/fluid/layers/nn.py:14815]: Done, thanks!

**Review by Heeenrrry — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

**Review by juncaipeng — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**